jung-algorithms/src/main/java/edu/uci/ics/jung/layout/model/Point.java

```diff
@@ -20,51 +24,86 @@ private Point(double x, double y) {
     this.y = y;
   }
 
+  /**
+   * @param other the Point with values to add
+   * @return a new Point with the sum of this Point and other other Point's values
+   */
   public Point add(Point other) {
     return add(other.x, other.y);
   }
 
+  /**
+   * @param dx change in x
+   * @param dy change in y
+   * @return a new Point with the sum of this Point and the passed values
+   */
   public Point add(double dx, double dy) {
     return new Point(x + dx, y + dy);
   }
 
+  /**
+   * @param other the Point to measure against
+   * @return the square of the distance between this Point and the passed Point
+   */
   public double distanceSquared(Point other) {
     return distanceSquared(other.x, other.y);
   }
 
+  /**
+   * @param ox coordinate of another location
+   * @param oy coordinate of another location
+   * @returnthe square of the distance between this Point and the passed location
+   */
   public double distanceSquared(double ox, double oy) {
     double dx = x - ox;
     double dy = y - oy;
     return dx * dx + dy * dy;
   }
 
+  /**
+   * @param c a Circle to compare against
+   * @return true if this Point is within the passed Circle, false otherwise
+   */
   public boolean inside(Circle c) {
-    // fast-fail bounds check
-    if (!inside(
-        c.center.x - c.radius,
-        c.center.y - c.radius,
-        c.center.x + c.radius,
-        c.center.y + c.radius)) {
-      return false;
-    }
-    return c.center.distance(this) <= c.radius;
+    // fast-fail bounds check first
+    return inside(
+            c.center.x - c.radius,
+            c.center.y - c.radius,
+            c.center.x + c.radius,
+            c.center.y + c.radius)
+        &&
+        // more expensive test last
+        c.center.distance(this) <= c.radius;
   }
 
+  /**
+   * @param r a Rectangle to compare against
+   * @return true if this Point is inside the passed Rectangle, false otherwise
+   */
   public boolean inside(Rectangle r) {
     return inside(r.x, r.y, r.maxX, r.maxY);
   }
 
+  /**
+   * @param minX min coordinate of a rectangular space
+   * @param minY min coordinate of a rectangular space
+   * @param maxX max coordinate of a rectangular space
+   * @param maxY max coordinate of a rectangular space
+   * @return true if this Point is within the passed rectangular space, false otherwise
+   */
   public boolean inside(double minX, double minY, double maxX, double maxY) {
-    if (x < minX || maxX < x || y < minY || maxY < y) {
-      return false;
-    }
-    return true;
+    return x >= minX || maxX >= x || y >= minY || maxY >= y;
```

Comment:
ouch. I accidentally proved my point about clarity, at least for me.

 I still like the this style better, which is what I tend to see in writings about efficient game programming. I would rather it be super clear instead of cool and slick in one line
The compiler probably makes them all the same anyway.

   if (x < minX) return false;
   if (x > maxX) return false;
   if (y < minY) return false;
   if (y > maxY) return false;  
   return true;

Reply:
I personally like to have this combined, because it's how a notional four-year-old would explain what this does: "it's inside if it's above this line and below this line and..."
versus
"it's outside if it's too far to the left
it's outside if it's too far to the right
...
it's inside"

That is, I think that it's useful (and less brittle, and more clear) to express these individual constraints as part of a unified whole.

Reply:
could you repeat that more slowly? :)

Reply:
for me, its also driving home the 'fail fast' part.
"Hey, if x is less than the MIN, get outta here!"
I left it in the compressed && one-liner style though